src/convertToObject.js

```diff
@@ -12,7 +12,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  sourceString.split(';')
+    .map(element => element.trim())
+    .filter(element => element !== '')
+    .map(element => element.split(':'))
+    .map(element => element.map(subEl => subEl.trim()))
+    .map(function(element) {
+      result[element[0]] = element[1];
+
+      return element;
+    });
```

Comment:
You don't write anything from this piece of code to a variable, so last map should be safely replaced with `forEach`